src/yunohost/dyndns.py

```diff
@@ -133,83 +172,53 @@ def dyndns_update(dyn_host="dyndns.yunohost.org", domain=None, key=None,
         ipv6 -- IPv6 address to send
 
     """
-    # IPv4
-    if ipv4 is None:
-        ipv4 = get_public_ip()
+    # Get old ipv4/v6
 
-    try:
-        with open('/etc/yunohost/dyndns/old_ip', 'r') as f:
-            old_ip = f.readline().rstrip()
-    except IOError:
-        old_ip = '0.0.0.0'
+    old_ipv4, old_ipv6 = (None, None)  # (default values)
 
-    # IPv6
-    if ipv6 is None:
-        try:
-            ip_route_out = subprocess.check_output(
-                ['ip', 'route', 'get', '2000::']).split('\n')
+    if os.path.isfile(OLD_IPV4_FILE):
+        old_ipv4 = read_file(OLD_IPV4_FILE).rstrip()
 
-            if len(ip_route_out) > 0:
-                route = IPRouteLine(ip_route_out[0])
-                if not route.unreachable:
-                    ipv6 = route.src_addr
```

Comment:
This method to get the ipv6 has the advantage to not depends from ip6.yunohost.org. But I am agree, it seems less sure than requesting ip6.yunohost.org